enumerable_methods.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i])
+    end
+  end
+
+  def my_each_with_index
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i], i)
+    end
+  end
+
+  def my_select
+    instance = self
+    result = []
+    instance.my_each do |e|
+      result.push(e) if yield(e)
+    end
+    result
+  end
+
+  def my_all?
+    instance = self
+    result = true
+    instance.my_each do |e|
+      unless yield(e)
+        result = false
+        break
+      end
+    end
+    result
+  end
+
+  def my_any?
+    instance = self
+    result = false
+    instance.my_each do |e|
+      if yield(e)
+        result = true
+        break
+      end
+    end
+    result
+  end
+
+  def my_none?
+    instance = self
+    result = true
+    instance.my_each do |e|
+      if yield(e)
+        result = false
+        break
+      end
+    end
+    result
+  end
+
+  def my_count
+    instance = self
+    instance.length
+  end
+
+  def my_map
+    instance = self
+    result = []
+    instance.my_each do |e|
+      result.push(yield(e))
+    end
+    result
+  end
```

Comment:
* [ ] if no block is given (i.e check for `block_given?` at the beginning of the method) the method should return an Enumerator

example: 
```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_map
     #LocalJumpError:
     #   no block given (yield)
      # ./enumerable_methods.rb:72:in `block in my_map'
      # ./enumerable_methods.rb:7:in `block in my_each'
      # ./enumerable_methods.rb:6:in `each'
      # ./enumerable_methods.rb:6:in `my_each'
      # ./enumerable_methods.rb:71:in `my_map'
  ```